pkg/fetcher/definition_test.go

```diff
@@ -87,6 +91,133 @@ func TestFindMatching(t *testing.T) {
 
 	})
 
+	t.Run("return tests selected by label expressions", func(t *testing.T) {
```

Comment:
Such test is not testing if the matching labels logic works fine, see the comment below line 77. The test is aware of the implementation. I'd like to see a black box test instead of the current one.

Reply:
It is changed now.